packages/editor/src/components/post-author/post-author-select.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+import { Component } from '@wordpress/element';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { SelectControl } from '@wordpress/components';
+
+export class PostAuthorSelect extends Component {
```

Comment:
Should we make it a functional component instead. It seems that it doesn't really need to be a class.

Reply:
And maybe take this as an opportunity to use useSelect/useDispatch iinstead.

Reply:
Sure, I brought this in unaltered as it existed previously to make sure it worked as expected. A simple refactor as a functional component would probably make this easier to understand. I'll work on it.

Reply:
Done! Refactored PostAuthorSelect as a functional component, switching to `useSelect` and `useDispatch`.